aesara/tensor/random/basic.py

```diff
@@ -1073,12 +1253,46 @@ def rng_fn_scipy(cls, rng, b, loc, scale, size):
 
 
 class BernoulliRV(ScipyRandomVariable):
+    r"""A Bernoulli discrete random variable.
+
+    The probability mass function for `bernoulli` in terms of the probability
+    of success :math:`p` of a single trial is:
+
+
+    .. math::
+
+        \begin{split}
+            f(k; p) = \begin{cases}
+                                (1-p)\quad \text{if $k = 0$},\\
+                                p\quad \text{if $k=1$}\\
+                        \end{cases}
+        \end{split}
+
+    where :math:`0 \leq p \leq 1`.
+
+    """
     name = "bernoulli"
     ndim_supp = 0
     ndims_params = [0]
     dtype = "int64"
     _print_name = ("Bern", "\\operatorname{Bern}")
 
+    def __call__(self, p, size=None, **kwargs):
+        r"""Draw samples from a Bernoulli distribution.
+
+        Parameters
+        ----------
+        p
+            Probability of success :math:`p` of a single trial.
+        size
+            Sample shape. If the given size is, e.g. `(m, n, k)` then `m * n * k`
+            independent, identically distributed random variables are
+            returned. Default is `None` in which case a single random variable
+            is returned.
```

Comment:
```suggestion
            independent, identically distributed samples are
            returned. Default is `None` in which case a single sample
            is returned.
```

Reply:
Note to self: check the broadcasting behavior.

Reply:
This is not correct, if `size=None` then `p.size` samples are returned in an array of shape `np.shape(p)`.

```python
import aesara.tensor as at

srng = at.random.RandomStream(0)
p = at.matrix()
rv = srng.bernoulli(p)

print(rv.eval({p: [[0.9, 0.7], [.1, .5]]}))
# [[0, 1], [0, 0]]
```

Reply:
Yes, `size=None` is equivalent to `size=p.shape`. How does NumPy describe it?

Reply:
`BernoulliRV` uses Scipy's sampling function, and Scipy just says "default computed from input arguments". I find that NumPy is inconsistent in the way they describe the samples' shape; we need to clarify what we want to document here (see my last comment in the main thread). I think that shape of the sample when arguments are in their most simple form (scalar, vector, ...) is probably best.